File: lib/go/internal/mock/PartiallyMatches.go
//|
//|  Copyright The Telepact Authors
//|  SPDX-License-Identifier: Apache-2.0
//|

package mock

// PartiallyMatches determines if any element in wholeList matches partElement using sub-map comparison semantics.
func PartiallyMatches(wholeList []any, partElement any) bool {
	for _, wholeElement := range wholeList {
		if IsSubMapEntryEqual(partElement, wholeElement) {
			return true
		}
	}
	return false
}

File: lib/ts/src/internal/mock/PartiallyMatches.ts
//|
//|  Copyright The Telepact Authors
//|  SPDX-License-Identifier: Apache-2.0
//|

import { isSubMapEntryEqual } from './IsSubMapEntryEqual.js';

export function partiallyMatches(wholeList: any[], partElement: any): boolean {
    for (const wholeElement of wholeList) {
        if (isSubMapEntryEqual(partElement, wholeElement)) {
            return true;
        }
    }
    return false;
}
